FWCore/AbstractServices/interface/RandomNumberGenerator.h

```diff
@@ -146,9 +146,11 @@ namespace edm {
 
   class RandomNumberGenerator {
   public:
-    RandomNumberGenerator() {}
+    RandomNumberGenerator() = default;
     RandomNumberGenerator(RandomNumberGenerator const&) = delete;
-    RandomNumberGenerator const& operator=(RandomNumberGenerator const&) = delete;
+    RandomNumberGenerator& operator=(RandomNumberGenerator const&) = delete;
+    RandomNumberGenerator(RandomNumberGenerator&&) = delete;
+    RandomNumberGenerator& operator=(RandomNumberGenerator&&) = delete;
     virtual ~RandomNumberGenerator();
```

Comment:
For my education, why arent we doing 
```
~RandomNumberGenerator() = default;
```
here instead of the .cc file?

Reply:
To have the vtable generated only once in a well-defined object (via an out-of-line virtual function definition). This was very relevant once upon a time, although I'm not sure how relevant it is nowadays. Certainly doesn't hurt (and thus I don't see a reason to deviate from the pattern that existed).